uxarray/grid.py

```diff
@@ -100,9 +100,7 @@ def __init__(self, dataset, **kwargs):
         self.__init_grid_var_attrs__()
 
         # construct connectivity
-        if self.source_grid != "From vertices":
-            if "Mesh2_edge_nodes" not in self.ds:
-                self._build_edge_node_connectivity()
+        self._build_edge_node_connectivity()
```

Comment:
Apologies for the late review, finals season has been extremely busy.

Let's include a new `kwarg` in our `Grid` constructor to indicate whether we want to build connectivities, perhaps call it `build_connectivity`. If it is `True`, we should construct all supported connectivtities.  Additionally, let's create a helper function called `Grid.build_grid_connectivity` that constructs all of our supported connectivities, which will both be called by the constructor (if the previous case is true) or if the user desires to construct the connectivtiies after initializing a `Grid`. This implementation should remove the need for the `repopulate` flag, however we can keep it for testing purposes given it's an internal function. If we chose to expose these individual construction functions to the user, we can create a wrapper with additional safeguards. Below is a rough idea of what I'm thinking 

```Python
kwargs_list = [
            'gridspec', 'vertices', 'islatlon', 'concave', 'source_grid',
            'use_dual', `build_connectivity`
        ]
```

```Python
# construct all supported connectivities, overriding existing parsed connectivities
if self.build_connectivity is not None and self.build_connectivity is True:
    self.build_grid_connectivity()
```
```Python
def build_grid_connectivity(sef):
    self._build_edge_node_connectivity()
    self._build_face_edges_connectivity()
    # any other connectivities will be added here
```

Reply:
@erogluorhan 

I was considering this type of approach, except instead of having separate arguments for each connectivity that we would like to build, we could have `build_connectivity` either be boolean for building all of them, or a list of strings if we want to build a subset of them. 

I'm still favoring keeping this PR simple enough to get merged ASAP and handling this issue elsewhere, and I am leaning in the direction of a boolean `build_connectivity` with public helper functions for building each individual connectivity with safeguards when calling the internal build functions.